api/src/resources/dataset.rs

```diff
@@ -17,7 +17,7 @@ use std::{
     str::FromStr,
 };
 
-#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
```

Comment:
To avoid having to remove the `Eq`, you should use `ordered_float::NotNan<f64>` which we already use for this purpose.